src/http/zcl_abapgit_ajson.clas.testclasses.abap

```diff
@@ -0,0 +1,1191 @@
+    cl_abap_unit_assert=>assert_equals(
+      act = lcl_utils=>normalize_path( '/abc' )
+      exp = '/abc/' ).
+    cl_abap_unit_assert=>assert_equals(
+      act = lcl_utils=>normalize_path( 'abc/' )
+      exp = '/abc/' ).
+    cl_abap_unit_assert=>assert_equals(
+      act = lcl_utils=>normalize_path( '/abc/' )
+      exp = '/abc/' ).
+
+  ENDMETHOD.
+
+  METHOD split_path.
+
+    DATA ls_exp TYPE zcl_abapgit_ajson=>ty_path_name.
+    DATA lv_path TYPE string.
+
+    lv_path     = ''. " alias to root
+    ls_exp-path = ''.
+    ls_exp-name = ''.
+    cl_abap_unit_assert=>assert_equals(
+      act = lcl_utils=>split_path( lv_path )
+      exp = ls_exp ).
+
+    lv_path     = '/'.
+    ls_exp-path = ''.
+    ls_exp-name = ''.
+    cl_abap_unit_assert=>assert_equals(
+      act = lcl_utils=>split_path( lv_path )
+      exp = ls_exp ).
+
+    lv_path     = '/abc/'.
+    ls_exp-path = '/'.
+    ls_exp-name = 'abc'.
+    cl_abap_unit_assert=>assert_equals(
+      act = lcl_utils=>split_path( lv_path )
+      exp = ls_exp ).
+
+    lv_path     = 'abc'.
+    ls_exp-path = '/'.
+    ls_exp-name = 'abc'.
+    cl_abap_unit_assert=>assert_equals(
+      act = lcl_utils=>split_path( lv_path )
+      exp = ls_exp ).
+
+    lv_path     = '/abc'.
+    ls_exp-path = '/'.
+    ls_exp-name = 'abc'.
+    cl_abap_unit_assert=>assert_equals(
+      act = lcl_utils=>split_path( lv_path )
+      exp = ls_exp ).
+
+    lv_path     = 'abc/'.
+    ls_exp-path = '/'.
+    ls_exp-name = 'abc'.
+    cl_abap_unit_assert=>assert_equals(
+      act = lcl_utils=>split_path( lv_path )
+      exp = ls_exp ).
+
+    lv_path     = '/abc/xyz'.
+    ls_exp-path = '/abc/'.
+    ls_exp-name = 'xyz'.
+    cl_abap_unit_assert=>assert_equals(
+      act = lcl_utils=>split_path( lv_path )
+      exp = ls_exp ).
+
+    lv_path     = '/abc/xyz/'.
+    ls_exp-path = '/abc/'.
+    ls_exp-name = 'xyz'.
+    cl_abap_unit_assert=>assert_equals(
+      act = lcl_utils=>split_path( lv_path )
+      exp = ls_exp ).
+
+  ENDMETHOD.
+
+ENDCLASS.
+
+**********************************************************************
+* READER
+**********************************************************************
+
+CLASS ltcl_reader_test DEFINITION FINAL
+  FOR TESTING
+  RISK LEVEL HARMLESS
+  DURATION SHORT.
+
+  PRIVATE SECTION.
+
+    METHODS get_value FOR TESTING RAISING zcx_abapgit_ajson_error.
+    METHODS exists FOR TESTING RAISING zcx_abapgit_ajson_error.
+    METHODS value_integer FOR TESTING RAISING zcx_abapgit_ajson_error.
+    METHODS value_number FOR TESTING RAISING zcx_abapgit_ajson_error.
+    METHODS value_boolean FOR TESTING RAISING zcx_abapgit_ajson_error.
+    METHODS value_string FOR TESTING RAISING zcx_abapgit_ajson_error.
+    METHODS members FOR TESTING RAISING zcx_abapgit_ajson_error.
+    METHODS slice FOR TESTING RAISING zcx_abapgit_ajson_error.
+    METHODS array_to_string_table FOR TESTING RAISING zcx_abapgit_ajson_error.
+    METHODS get_date FOR TESTING RAISING zcx_ajson_error.
```

Comment:
Seems that `zcx_ajson_error` is missing?
![image](https://user-images.githubusercontent.com/17437789/89099486-ea505880-d3ef-11ea-8f2e-87e6d4838575.png)

Reply:
Should be `ZCX_ABAPGIT_AJSON_ERROR`?

Reply:
yeap, good catch, thanks. Fixed

Reply:
https://github.com/abaplint/abaplint/issues/1117 opened to catch this kind of error in the future